shadowsocks/stream.go

```diff
@@ -88,28 +91,43 @@ func (sw *shadowsocksWriter) Write(p []byte) (int, error) {
 	return int(n), err
 }
 
+func isZero(b []byte) bool {
+	for _, v := range b {
+		if v != 0 {
+			return false
+		}
+	}
+	return true
+}
+
 func (sw *shadowsocksWriter) ReadFrom(r io.Reader) (int64, error) {
 	if err := sw.init(); err != nil {
 		return 0, err
 	}
 	var written int64
-	sizeBuf := sw.buf[:2+sw.aead.Overhead()]
-	payloadBuf := sw.buf[len(sizeBuf):]
+
+	// sw.buf starts with the salt.
+	saltSize := sw.ssCipher.SaltSize()
+	// Normally we ignore the salt at the beginning of sw.buf.
```

Comment:
Let's add a comment explaining that we write the salt with the first message so that the first message doesn't have a distinctive size.

Reply:
Done